app/agent/services/volumeBrowserService.js

```diff
@@ -1,27 +1,37 @@
 angular.module('portainer.agent').factory('VolumeBrowserService', [
-  '$q', 'Browse',
-  function VolumeBrowserServiceFactory($q, Browse) {
+  'StateManager', 'Browse', 'BrowseVersion1',
+  function VolumeBrowserServiceFactory(StateManager, Browse, BrowseVersion1) {
     'use strict';
     var service = {};
 
+    function getAgentApiVersion() {
+      var state = StateManager.getState();
+      return state.endpoint.agentApiVersion;
+    }
+
+    function getBrowseService() {
+      var agentVersion = getAgentApiVersion();
+      return agentVersion > 1 ? Browse : BrowseVersion1;
+    }
+
     service.ls = function(volumeId, path) {
-      return Browse.ls({ volumeID: volumeId, path: path }).$promise;
+      return getBrowseService().ls({ volumeID: volumeId, path: path, version: getAgentApiVersion() }).$promise;
```

Comment:
See my point above, might need to declare `/browse/:volumeID/:action` in the rest service.

Reply:
this is fixed with the change in the version 1 browse service